### PR TITLE
Extract PatientSearchResultViewHolder to different file

### PR DESCRIPTION
**Pull request by vinaysshenoy**
Pivotal Story Link: https://www.pivotaltracker.com/story/show/166811630

This was actually done as a part of #702, but was broken out for easier review.

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/widgets/PatientSearchResultView.kt [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+  @field:Inject
+  lateinit var utcClock: UtcClock
+
+  @field:Inject
+  lateinit var userClock: UserClock
+
+  override fun onFinishInflate() {
+    super.onFinishInflate()
+    TheActivity.component.inject(this)
```
Comment (rakshakhegde) [inline app/src/main/java/org/simple/clinic/widgets/PatientSearchResultView.kt]: `PatientSearchResultView` right now assumes and depends on being used in `list_patient_search.xml`. As you are trying to make this view separate, can you inject the view here? Please refer `PatientSearchView#L53`.

File: app/src/main/res/layout/list_patient_search.xml [hunk 2]
```diff
@@ -19,7 +20,7 @@
     android:paddingTop="@dimen/spacing_16">
```
Comment (rakshakhegde) [inline app/src/main/res/layout/list_patient_search.xml]: As per above comment, you can put this `LinearLayout` in another XML file and inflate it from the `PatientSearchResultView`.

File: app/src/main/java/org/simple/clinic/widgets/PatientSearchResultView.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/widgets/PatientSearchResultView.kt]: Done.

File: app/src/main/res/layout/list_patient_search.xml (hunk 2 above)
Comment (vinaysshenoy) [reply app/src/main/res/layout/list_patient_search.xml]: Done.

**Pull request merged**